plugins/filter_lua/lua.c

```diff
@@ -418,7 +418,18 @@ static int cb_lua_filter(const void *data, size_t bytes,
         lua_pushstring(ctx->lua->state, tag);
         lua_pushnumber(ctx->lua->state, ts);
         lua_pushmsgpack(ctx->lua->state, p);
-        lua_call(ctx->lua->state, 3, 3);
+        if (ctx->protected_mode) {
+            ret = lua_pcall(ctx->lua->state, 3, 3, 0);
+            if (ret != 0) {
+                flb_plg_error(ctx->ins, "error code %d: %s",
+                              ret, lua_tostring(ctx->lua->state, -1));
+                lua_pop(ctx->lua->state, 1);
+                return FLB_FILTER_NOTOUCH;
```

Comment:
exiting at this point is leaking memory, take a look at the valgrind output. 

the sbuffer must be destroyed.